stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/viewholder/internal/MediaAttachmentsViewHolder.kt

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.ui.message.list.adapter.viewholder.internal
+
+import android.view.ViewGroup
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.core.view.isVisible
+import androidx.core.view.setPadding
+import androidx.core.view.updateLayoutParams
+import com.getstream.sdk.chat.adapter.MessageListItem
+import io.getstream.chat.android.client.models.Attachment
+import io.getstream.chat.android.ui.R
+import io.getstream.chat.android.ui.common.extensions.internal.dpToPx
+import io.getstream.chat.android.ui.common.extensions.internal.streamThemeInflater
+import io.getstream.chat.android.ui.common.internal.LongClickFriendlyLinkMovementMethod
+import io.getstream.chat.android.ui.databinding.StreamUiItemMessageMediaAttachmentBinding
+import io.getstream.chat.android.ui.message.list.adapter.MessageListItemPayloadDiff
+import io.getstream.chat.android.ui.message.list.adapter.MessageListListenerContainer
+import io.getstream.chat.android.ui.message.list.adapter.internal.DecoratedBaseMessageItemViewHolder
+import io.getstream.chat.android.ui.message.list.adapter.view.internal.AttachmentClickListener
+import io.getstream.chat.android.ui.message.list.adapter.view.internal.AttachmentLongClickListener
+import io.getstream.chat.android.ui.message.list.adapter.viewholder.decorator.internal.Decorator
+import io.getstream.chat.android.ui.transformer.ChatMessageTextTransformer
+
+/**
+ * ViewHolder used for displaying messages that contain image and/ or video attachments.
+ *
+ * @param parent The parent container.
+ * @param decorators List of decorators applied to the ViewHolder.
+ * @param listeners Listeners used by the ViewHolder.
+ * @param messageTextTransformer Formats strings and sets them on the respective TextView.
+ * @param binding Binding generated for the layout.
+ */
+internal class MediaAttachmentsViewHolder(
```

Comment:
The original might be internal as well but I opted for duplication instead of renaming here because it is used inside a protected function inside of the public class `MessageListItemViewHolderFactory` so it's used by customers via a public proxy function.